docs/DeveloperGuide.md

```diff
@@ -1,29 +1,739 @@
 # Developer Guide
 
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Budget Management
 
+#### Implementation
+The Budget Management feature allows users to set financial limits for the various categories and monitor their spending. 
+This feature's objective is to give users the ability to stay within their financial goals and avoid overspending.
 
-## Product scope
-### Target user profile
+This feature is orchestrated by `ListBudgetCommand` and `SetBudgetCommand`, which are initialised by the `Parser` 
+class. Below is a description of the key class attributes and methods involved in the budget setting and listing 
+process:
+
+##### Class Attributes for `SetBudgetCommand`:
+| Class Attribute | Variable Type | Relevance                                                           |
```

Comment:
Rendering errors for the command table in HTML view. Recommend checking the formatting for the tables again.